### ISSUE TITLE
How to pass variables from one module into another without import

### ISSUE THREAD

**Issue by paceaux** [sass/sass#3257]
Howdy, 

this is related, but not identical, to issue #2890. I've read the [Stack Overflow response](https://stackoverflow.com/questions/63361377/undefined-font-awesome-variable-when-use) but unfortunately, that's not satisfactory in my case. 

Fundamentally my goal is this:

* get variables from one module coming out of node_modules
* Pass those variables to another module coming from node_modules that is depending on those variables 

pre dart-sass this would've looked something like this:

```
@import '../../node_modules/font-awesome-sass/assets/stylesheets/font-awesome/variables';
@import  '../../node_modules/font-awesome-sass/assets/stylesheets/font-awesome/icons';
```

But what is the equivalent in _dart-sass_ ?

The `with` keyword [won't accept a variable that is _already_ a map as an argument](https://github.com/paceaux/sass-use-problems/blob/passing-variables/src/styles/objects/icons.scss)

[I can try using](https://github.com/paceaux/sass-use-problems/blob/using-meta-load/src/styles/objects/icons.scss) `meta.css-load` and the `with` argument, but that requires that I provide variables where I've removed the `$` from the keys.  I can't find from Sass documentation exactly how I could _map_ a `map`. 

[I have tried](https://github.com/paceaux/sass-use-problems/blob/using-module-variables-and-meta-load/src/styles/objects/icons.scss) using `meta.module-variables` as the value for `$with` in a `meta.load-css` case, **but that doesn't work either** .  it _seems_ like `meta.module-variables` was made specifically to work with `meta.load-css`, but instead I get "undefined variable"

[I've even tried](https://github.com/paceaux/sass-use-problems/blob/module-variables-meta-load-as-variable/src/styles/objects/icons.scss) setting the `meta.load-css`'s `with` argument with a _variable_ . The variable comes up undefined. 

In summary:

1. The `with` keyword can't accept a map
2. `load-css` specifically requires a map where the `$` isn't part of the variable name, but the documentation for this feature doesn't explain how to produce such a map
3. `load-css` paired with `module-variables` doesn't work, even though module-variables is supposed to remove the `$` from the keys
4. the looping documentation doesn't provide an example for rewriting keys

So what's the recommendation here? I feel like all of my attempts were reasonable and intuitive, but I'm happy to be told my intuition is wrong. If the documentation needs updating, can I help? What Would the Sass Folks Do?

**Comment by jathak** [comment]
Does `icons` define configurable variables? (declared with the `!default` flag) If not, you won't be able to override them with a with clause.

It looks like the `font-awesome-sass` NPM module hasn't been updated in 5 years (while the corresponding GitHub repo was last updated nearly 2 years ago), so it's likely the individual stylesheets in that package predate the module system and assume that they'll all be imported in a global context. Given that, you won't be able to properly depend on them with `@use`. You'll likely want to look for a more actively maintained package.

**Comment by paceaux** [comment]
Thank you @jathak . `icons` does use configurable variables that are declared with the `!default` flag. 

The error that I hit is `Error: Undefined variable.` And it's on _specifically_ a variable with a `default`

```
  ╷
4 │ .#{$fa-css-prefix}-glass:before { content: $fa-var-glass; }
  │    ^^^^^^^^^^^^^^
```

[Here's an example where I don't use font awesome](https://github.com/paceaux/sass-use-problems/blob/non-module-variables/src/styles/main.scss) _at all_ but I still get the same error. (and I was sure to set every variable with `!default`) It seems like `meta.load-css` has a problem accepting module variables. 

Could you please advise how I would get variables out of module A and into Module B, _or_ confirm that there's an error somewhere between `meta.load-css` and `meta.module-variables` ?
